File: customer-service/src/main/java/com/orbitbank/customerservice/controller/CustomerController.java
package com.orbitbank.customerservice.controller;

import com.orbitbank.customerservice.model.Customer;
import com.orbitbank.customerservice.service.CustomerService;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/orbit/customer/")
public class CustomerController {

    @Autowired
    private CustomerService customerService;

    @PostMapping("/")
    public Customer addCustomer(@RequestBody Customer customer){
        return customerService.save(customer);
    }

    @GetMapping("/")
    public List<Customer> getAllCustomers(){
        return customerService.getAllCustomers();
    }

    @GetMapping("/{id}")
    public Customer getCustomerById(@PathVariable("id") String customerId){
        return customerService.getCustomerById(customerId);
    }

    @PutMapping("/{id}")
    public Boolean updateCustomerDetails(@PathVariable("id") String customerId,@RequestBody Customer customerDetails){
        return customerService.updateCustomerDetails(customerId,customerDetails);
    }

    @PutMapping("/{id}")
    public Boolean disableCustomerAccount(@PathVariable("id") String customerId){
        return customerService.disableCustomerAccount(customerId);
    }

}

File: customer-service-kotlin/customer-service-kotlin/src/main/kotlin/com/orbitbank/customerservicekotlin/controller/CustomerController.kt
package com.orbitbank.customerservicekotlin.controller

import com.orbitbank.customerservicekotlin.model.Customer
import com.orbitbank.customerservicekotlin.service.CustomerService
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("/api/orbit/customer/")
class CustomerController {
    @Autowired
    private val customerService: CustomerService? = null
    @PostMapping("/")
    fun addCustomer(@RequestBody customer: Customer?): Customer {
        return customerService!!.saveCustomer(customer!!)
    }

    @get:GetMapping("/")
    val getAllCustomer: List<Any?>?
        get() {
            val support: List<Customer> = ArrayList<Customer>()
            return customerService?.listAllCustomers()
        }

    @GetMapping("/{id}")
    fun getCustomerById(@PathVariable("id") customerId: String?): Customer? {
        return customerService!!.getCustomerById(customerId)
    }

    @PutMapping("/{id}")
    fun updateCustomerDetails(
        @PathVariable("id") customerId: String?,
        @RequestBody customerDetails: Customer?
    ): Boolean {
        return customerService!!.updateCustomerDetails(customerId, customerDetails!!)
    }

    @PutMapping("/{id}")
    fun disableCustomerAccount(@PathVariable("id") customerId: String?): Boolean {
        return customerService!!.disableCustomerAccount(customerId)
    }
}